src/ensembl/src/content/app/entity-viewer/state/gene-view/entityViewerGeneActions.ts

```diff
@@ -0,0 +1,104 @@
+import { createAction } from 'typesafe-actions';
+import set from 'lodash/set';
+import cloneDeep from 'lodash/cloneDeep';
+import { ActionCreator, Action } from 'redux';
+import { ThunkAction } from 'redux-thunk';
+
+import {
+  getEntityViewerActiveGenomeId,
+  getEntityViewerActiveEnsObjectId
+} from 'src/content/app/entity-viewer/state/general/entityViewerGeneralSelectors';
+import { getEntityViewerActiveGenomeConfiguration } from 'src/content/app/entity-viewer/state/gene-view/entityViewerGeneSelectors';
+import { EntityViewerGeneObjectState } from 'src/content/app/entity-viewer/state/gene-view/entityViewerGeneState';
+
+import { RootState } from 'src/store';
+
+export const updateActiveGeneObjectState = createAction(
+  'entity-viewer/update-state-for-active-genome',
+  (payload: {
+    activeGenomeId: string;
+    activeObjectId: string;
+    data: EntityViewerGeneObjectState;
+  }) => {
+    const { activeGenomeId, activeObjectId, data } = payload;
+    return { activeGenomeId, activeObjectId, data: cloneDeep(data) };
+  }
+)();
+
+export const setActiveGeneTab: ActionCreator<ThunkAction<
+  void,
+  any,
+  null,
+  Action<string>
+>> = (selectedTabName: boolean) => (dispatch, getState: () => RootState) => {
+  const activeGenomeId = getEntityViewerActiveGenomeId(getState());
+  const activeObjectId = getEntityViewerActiveEnsObjectId(getState());
+
+  if (!activeGenomeId || !activeObjectId) {
+    return;
+  }
+
+  dispatch(
+    updateActiveGeneObjectState({
+      activeGenomeId,
+      activeObjectId,
+      data: set(
```

Comment:
I don't remember whether getState returns the actual state or the copy of the state (I assume the actual state, making a copy would be expensive). If it's the actual state, then you have just mutated the state, which is a big no-no. If it's the copy of the state (unlikely), then `cloneDeep` in `updateActiveGeneObjectState` has no purpose.

Reply:
I guess this no longer applies after addressing the previous comment